docs: Add usage sub-section in docs (#604)
awslabs/aws-lambda-powertools-typescript: docs/core/logger.md

npm install @aws-lambda-powertools/logger
```

### Usage

> The `Logger` utility should always be instantiated outside of the Lambda handler. In doing this, subsequent invocations processed by the same instance of your function can reuse these resources. This saves cost by reducing function run time. In addition to this, `Logger` can also be aware of things like whether or not a given invocation had a cold start or not and inject the appropriate fields into the log.

Reviewer: ```suggestion
The `Logger` utility must always be instantiated outside of the Lambda handler. In doing this, subsequent invocations processed by the same instance of your function can reuse these resources. This saves cost by reducing function run time. In addition, `Logger` can track cold start and inject the appropriate fields into the logs.
```
